frontend/src/container/NewWidget/RightContainer/Threshold/Threshold.tsx

```diff
@@ -0,0 +1,262 @@
+						thresholdUnit: unit,
+						thresholdValue: value,
+					};
+				}
+				return threshold;
+			}),
+		);
+	};
+
+	const editHandler = (): void => {
+		setIsEditMode(true);
+	};
+
+	const handleOperatorChange = (value: string | number): void => {
+		setOperator(value);
+	};
+
+	const handleValueChange = (value: number | null): void => {
+		if (value === null) {
+			return;
+		}
+		setValue(value);
+	};
+
+	const handleUnitChange = (value: string): void => {
+		setUnit(value);
+	};
+
+	const handlerFormatChange = (
+		value: ThresholdProps['thresholdFormat'],
+	): void => {
+		setFormat(value);
+	};
+
+	const deleteHandler = (): void => {
+		if (thresholdDeleteHandler) {
+			thresholdDeleteHandler(index);
+		}
+	};
+
+	const ref = useRef<HTMLDivElement>(null);
+	const [{ handlerId }, drop] = useDrop<
+		ThresholdProps,
+		void,
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		{ handlerId: any }
+	>({
+		accept: 'Threshold',
+		collect(monitor) {
+			return {
+				handlerId: monitor.getHandlerId(),
+			};
+		},
+		hover(item: ThresholdProps, monitor) {
+			if (!ref.current) {
+				return;
+			}
+			const dragIndex = item.keyIndex;
+			const hoverIndex = keyIndex;
+
+			if (dragIndex === hoverIndex) {
+				return;
+			}
+
+			const hoverBoundingRect = ref.current?.getBoundingClientRect();
+
+			const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+
+			const clientOffset = monitor.getClientOffset();
+
+			const hoverClientY = (clientOffset as XYCoord).y - hoverBoundingRect.top;
+
+			if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+				return;
+			}
+
+			if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+				return;
+			}
+
+			moveThreshold(dragIndex, hoverIndex);
+			// eslint-disable-next-line no-param-reassign
+			item.keyIndex = hoverIndex;
+		},
+	});
+
+	const [{ isDragging }, drag] = useDrag({
+		type: 'Threshold',
+		item: () => ({ keyIndex }),
+		collect: (monitor) => ({
+			isDragging: monitor.isDragging(),
+		}),
+	});
+
+	const opacity = isDragging ? 0 : 1;
+	drag(drop(ref));
+
+	return (
+		<div
+			ref={ref}
+			style={{ opacity }}
+			data-handler-id={handlerId}
+			className="threahold-container"
+		>
+			<Card
+				className={
+					isDarkMode
+						? `threahold-card threahold-card-dark`
+						: `threahold-card threahold-card-light`
+				}
+			>
+				<div className="threshold-card-container">
+					<div className="threshold-action-button">
+						{isEditMode ? (
+							<CheckOutlined onClick={saveHandler} />
+						) : (
+							<EditOutlined className="threshold-action-icon" onClick={editHandler} />
+						)}
+						<Divider type="vertical" />
+						<DeleteOutlined
+							className="threshold-action-icon"
+							onClick={deleteHandler}
+						/>
+					</div>
+					<div>
+						<Space>
+							<Typography.Text>If value is</Typography.Text>
+							{isEditMode ? (
+								<Select
+									style={{ minWidth: '73px', backgroundColor: '#141414' }}
```

Comment:
why do we specifically have the minWidth to `73px`?

Reply:
Udpated it to max Width